src/scheduler/workers/main.new.ts

```diff
@@ -0,0 +1,26 @@
+import { main as unpublishWeeklyTask } from "./unpublish/weekly";
+import { main as unpublishDayBeforeTask } from "./unpublish/dayBefore";
+import { main as unpublishDayTask } from "./unpublish/day";
+import { processAnnualReview } from "./main";
+import { logger } from "server/services/logger";
+
+async function main() {
+  await processAnnualReview()
+  // put all worker tasks to be executed here. They will be executed async (non blocking/non sequential).
+  // const tasks = [unpublishWeeklyTask(), unpublishDayBeforeTask(), unpublishDayTask()];
+  // @todo UNCOMMENT this once finished testig
+  // return await processAnnualReview();
+
+  const tasks = [unpublishDayTask()];
+
+  return await Promise.allSettled(tasks);
```

Comment:
I could be wrong but for `Promise.allSettled` do you execute the function in the array or do you just pass the function name? As in:
```js
const tasks = [unpublishDayTask];
```

instead of:
```js
const tasks = [unpublishDayTask()];
```

Reply:
Promise.allSettled takes a single or array of promises.  As long as the function returns the promises you can execute the function call directly in it.  I think in this case there's an argument for calling the function directly in Promise.allSettled.  Thanks for spotting.